.github/workflows/vast.yaml

```diff
@@ -1079,23 +607,448 @@ jobs:
+
+      - name: Upload Artifact to GitHub
+        uses: actions/upload-artifact@v3
+        with:
+          name: "${{ env.PACKAGE_NAME }}.tar.gz"
+          path: "${{ env.BUILD_DIR }}/package/${{ env.PACKAGE_NAME }}.tar.gz"
+          if-no-files-found: error
+
+      - name: Configure GCS Credentials
+        if: ${{ github.event_name == 'push' || github.event_name == 'release' }}
+        uses: google-github-actions/setup-gcloud@v0
+        with:
+          service_account_key: ${{ secrets.GCP_SA_KEY }}
+          export_default_credentials: true
+
+      - name: Upload Artifact to GCS
+        if: ${{ github.event_name == 'push' || github.event_name == 'release' }}
+        run: |
+          gsutil -m cp "${{ env.BUILD_DIR }}/package/${{ env.PACKAGE_NAME }}.tar.gz" "gs://${{ secrets.GCS_BUCKET }}/${{ env.PACKAGE_NAME }}.tar.gz"
+
+      # This step ensures that assets from previous runs are cleaned up to avoid
+      # failure of the next step (asset upload)
+      - name: Delete existing Release Assets
+        if: github.event_name == 'release'
+        uses: mknejp/delete-release-assets@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          tag: ${{ github.ref }}
+          # don't fail if no previous assets exist.
+          fail-if-no-assets: false
+          # only delete assets when `tag` refers to a release
+          fail-if-no-release: tru
+          assets: "${{ env.PUBLISH_NAME }}.tar.gz"
+
+      - name: Publish to GitHub Release
+        if: github.event_name == 'release'
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ github.event.release.upload_url }}
+          asset_path: "${{ env.BUILD_DIR }}/package/${{ env.PACKAGE_NAME }}.tar.gz"
+          # The asset names are constant so we can permanently link to
+          # https://github.com/tenzir/vast/releases/latest/download/vast-debian-release-gcc.tar.gz
+          # https://github.com/tenzir/vast/releases/latest/download/vast-debian-release-clang.tar.gz
+          # for builds of the latest release.
+          asset_name: "${{ env.PUBLISH_NAME }}.tar.gz"
+          asset_content_type: application/gzip
+
+      - name: Publish VAST.spdx to GitHub Release
+        if: ${{ github.event_name == 'release' && matrix.vast.name == 'Debian' }}
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ github.event.release.upload_url }}
+          asset_path: "VAST.spdx"
+          asset_name: "VAST.spdx"
+          asset_content_type: text/plain
+
+  vast-plugins:
+    needs:
+      - vast
+      - configure
+    if: ${{ always() && needs.configure.outputs.run-vast-plugins == 'true' }}
+    runs-on: ${{ matrix.plugin.os }}
+    container: ${{ matrix.plugin.container }}
+    strategy:
+      fail-fast: false
+      matrix:
+        plugin:
```

Comment:
If you add a second record on this level you can make the matrix two-dimensional. I'm not sure whether that makes much sense, but it is an option here.

Reply:
Definitely, less repetition. I will try to get this working